### PR TITLE
CATTY-101: Open project from URL

### PR DESCRIPTION
**Pull request by afibian**
Due to changes in iOS, it was no longer possible to import projects from share.catrob.at
This PR implements apples newer [universal links] for importing projects

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

[universal links]: https://developer.apple.com/documentation/uikit/inter-process_communication/allowing_apps_and_websites_to_link_to_your_content

### CODE REVIEW

**Comment by afibian** [comment]
@KrennKristof To be honest I'm also not 100% sure about that either.
However, the core functionality (downloading and saving the project) is still performed by the existing implementation in `CBFileManager`. 
What this PR adds is an interface for iOS to open certain links directly in our app. Unfortunately testing this behavior is tricky, as a lot of this happen outside of our application.

**Comment by KrennKristof** [comment]
I will test it on my device a few times, but as I already said, it looks good to me

**Comment by afibian** [comment]
Perfect, thank you :)
And if you have any ideas on how to add meaningful tests to this PR please let me know, I'll be happy to implement them.

File: src/Catty.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -11556,14 +11576,18 @@
 				INFOPLIST_FILE = "Catty/Supporting Files/App-Info.plist";
 				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
+				LIBRARY_SEARCH_PATHS = (
```
Comment (m-herold) [inline src/Catty.xcodeproj/project.pbxproj:11579]: Why is this change necessary?

File: src/Catty/ViewController/Download/ChartProjects/ChartProjectsStoreViewController.swift [hunk 2]
```diff
@@ -81,27 +81,6 @@ class ChartProjectsStoreViewController: UIViewController, SelectedChartProjectsD
 
     // MARK: - Helper Methods
 
-    private func mapStoreProjectToCatrobatProject(project: StoreProject) -> CatrobatProject {
```
[the remark this answers is not in the document]
Comment (afibian) [reply src/Catty/ViewController/Download/ChartProjects/ChartProjectsStoreViewController.swift:84]: Done

File: src/Catty/DataStructures/JSON/StoreProject.swift [hunk 3]
```diff
@@ -105,4 +105,25 @@ struct StoreProject: Codable {
         fileSize = try container.decodeIfPresent(Float.self, forKey: .fileSize)
         featuredImage = try container.decodeIfPresent(String.self, forKey: .featuredImage)
     }
+
+    func toCatrobatProject() -> CatrobatProject {
```
[the remark this answers is not in the document]
Comment (afibian) [reply src/Catty/DataStructures/JSON/StoreProject.swift:109]: Thank you :)

File: src/Catty/ViewController/CatrobatTableViewController/CatrobatTableViewController+OpenURL.swift [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            let storyboard = UIStoryboard(name: "iPhone", bundle: nil)
+            guard let viewController = storyboard.instantiateViewController(withIdentifier: "ProjectDetailStoreViewController") as? ProjectDetailStoreViewController else { return }
+            viewController.project = catrobatProject
+//            self.performSegue(withIdentifier: kSegueToProjectDetail, sender: self)
+            self.navigationController?.pushViewController(viewController, animated: true)
+        })
+    }
+}
+
+func idFromURL(url: URL) -> String? {
```
[the remark this answers is not in the document]
Comment (afibian) [reply src/Catty/ViewController/CatrobatTableViewController/CatrobatTableViewController+OpenURL.swift]: Done

File: src/Catty/ViewController/Download/ChartProjects/ChartProjectsStoreViewController.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (afibian) [reply src/Catty/ViewController/Download/ChartProjects/ChartProjectsStoreViewController.swift:84]: Done

File: src/Catty/DataStructures/JSON/StoreProject.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (afibian) [reply src/Catty/DataStructures/JSON/StoreProject.swift:109]: Thank you :)

File: src/CattyTests/ViewController/CatrobatTableViewController+OpenURLTest.swift .swift [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+ */
+
+import Foundation
+import XCTest
+
+@testable import Pocket_Code
+
+final class CatrobatTableViewControllerOpenURLTests: XCTestCase {
+
+    let testId = "817"
```
[the remark this answers is not in the document]
Comment (afibian) [reply src/CattyTests/ViewController/CatrobatTableViewController+OpenURLTest.swift .swift]: Done

File: src/Catty.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -3764,6 +3770,14 @@
 /* End PBXFrameworksBuildPhase section */
 
 /* Begin PBXGroup section */
+		05A3D46624F517220051DB39 /* DataStructures */ = {
```
[the remark this answers is not in the document]
Comment (afibian) [reply src/Catty.xcodeproj/project.pbxproj]: Done

File: src/Catty.xcodeproj/project.pbxproj (hunk 1 above)
Comment (afibian) [reply src/Catty.xcodeproj/project.pbxproj:11579]: Sorry this change slipped through, my best guess is its a reminiscence of an old develop branch. I Deleted this change.

**Review by m-herold — APPROVED** [review]
